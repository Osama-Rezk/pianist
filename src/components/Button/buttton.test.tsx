import React from "react";

import { Button } from "./";
import { shallow } from "enzyme";

describe("Button Component", () => {
  it("should render my component", () => {
    const wrapper = shallow(<Button title="Test Button" onClick={() => {}} />);

    expect(wrapper.exists()).toBe(true);
  });

  it("should render correctly", () => {
    const wrapper = shallow(<Button title="Test Button" onClick={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("it should not return null", () => {
    const wrapper = shallow(<Button title="Test Button" onClick={() => {}} />);

    expect(wrapper.isEmptyRender()).toBe(false);
  });

  it("it should Render Title ", () => {
    const wrapper = shallow(<Button title="Test Button" onClick={() => {}} />);

    expect(wrapper.props().children).toEqual("Test Button");
  });

  it("should call mock function when button is clicked", () => {
    const mockFn = jest.fn();

    const wrapper = shallow(<Button title="Test Button" onClick={mockFn} />);
    wrapper.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });
});
