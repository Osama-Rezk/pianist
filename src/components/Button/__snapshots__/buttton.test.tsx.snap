// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render my component 1`] = `ShallowWrapper {}`;
